### ISSUE TITLE
VAEGAN training failing

### ISSUE THREAD

**Issue by vishwakftw** [DL-IT/generative_zoo#1]
The VAEGAN training fails after about 30 iterations.

**Comment by vishwakftw** [comment]
The training is stabilized. The L<sup>th</sup> layer loss is decreasing, but very slowly. The images are incomprehensible.

**Comment by vishwakftw** [comment]
I think the issue has been fixed. The problem was with the loss function specified in the paper. L<sub>{gan}</sub> is supposed to be of the opposite sign since we are minimizing it for the discriminator, as specified in the algorithm. The code was written verbatim from the text, hence the problem.

**Comment by vishwakftw** [comment]
However, after proposing the above fix, the image that are learned are incomprehensible. Below is the graph of the variation of all losses with the iterations.

![vaegan_train](https://user-images.githubusercontent.com/23639302/28803496-a0cd775a-764c-11e7-9dcd-4863d099ed4a.png)

Optimizer used is Adam with learning rate: 0.0005 and (beta<sub>1</sub>, beta<sub>2</sub>) = (0.5, 0.9)

Any help would be appreciated.
